### ISSUE TITLE
Bugs that are fixed in electron-devtools-installer are not fixed in vue-cli-plugin-electron-builder

### ISSUE THREAD

**Issue by motin** [nklayman/vue-cli-plugin-electron-builder#91]
**TL;DR:** Don't bundle a copy of electron-devtools-installer. Instead: Create a PR with the fix [(as suggested elsewhere)](https://github.com/nklayman/vue-cli-plugin-electron-builder/pull/35#issuecomment-407313249) and if custom modifications are necessary, push them to the master branch of the fork, then use the forked version of electron-devtools-installer as a dependency. This way, it is possible for app developers to use newer upstreams versions (or forked custom versions) of electron-devtools-installer without having to clone, patch and rebuild vue-cli-plugin-electron-builder.

**Describe the bug**
The package electron-devtools-installer is not used. Instead, a modified copy of the package is bundled. 
Among other dependency admin nightmares this causes, bugs that will be fixed in electron-devtools-installer recently are not automatically fixed in vue-cli-plugin-electron-builder

**To Reproduce**
1. Have the bad fortune to run into some bug fixed upstream and yet not manually ported to vue-cli-plugin-electron-builder
2. Install the latest electron-devtools-installer package via `npm install --save-dev MarshallOfSound/electron-devtools-installer#master`

**Expected behavior**
The bug should not manifest in the Electron app.

**Actual behavior**
The bug still manifests in the Electron app. 
Debugging the code becomes confusing since there are multiple versions of the electron-devtools-installer code available in node_modules and it is not clear which one is used.

**Environment:**
 - vue-cli-plugin-electron-builder version : 1.0.0-rc.3

**Comment by nklayman** [comment]
First, thank you for submitting the PR to electron-devtools-installer, I totally forgot about that :+1: .

However, I don't think it is an issue that I package my own version of electron-devtools installer. While your point about bug fixes not being included is valid, in this case it isn't really an issue. The last release of electron-devtools-installer was 4 months ago, and that was 5 months after the second-latest version. I am watching the repo, and if any changes get made, I will incorporate them into vcp-electron-builder. In fact, the custom version is actually more up-to-date then the published electron-devtools-installer (it has commits that haven't been published to npm). As for the user wanting to use a custom version of electron-devtools-installer, this is an edge case scenario, and nothing stops the user from installing and using it manually.

Sorry if I am shutting down your idea, and I agree with some of your points. It's just that, in my opinion, I feel like the current solution is the best option for the average user. If you feel like I didn't answer some of your concerns, feel free to reopen the issue and let me know what I missed.

**Comment by motin** [comment]
TL;DR - for the sake of a healthy ecosystem and less headaches for contributors and users in the long term, please do not bundle modified versions of other packages - at least not without making it crystal clear why it has been done and where. 

I understand that in this particular case, in this particular point in time, it may seem better to bundle a custom-modified version of another package since electron-devtools-installer has had very little activity while at the same time you have been very active. Also, there is literally only one package that is bundled+modified here. Thus there are few upstream fixes and there is little lag between a bug fix upstream and the corresponding fix in this repository. 

Despite this, already in this particular case I had to backtrack quite a bit to understand the situation, find the reasoning, detect and fix the underlying library, send a PR, attempt to bundle the fix into this package, find that the fix was actually ported by you already, but without a reference to any issue, etc. Took some time from an afternoon I'd rather spend time with other things on. 

Then imagine that you become busy with other things in life preventing you from backporting the bugs quickly and/or more bugs gets fixed in the other package. Consumers of and contributors to this and the bundled package then get affected more often. 

(The reason why I am writing this comment (which is more about open source ethics / philosophy in general than the current situation) is that I personally get bitten by this all the time in various packages - just trying to find myself in an ecosystem in the future where I do not have to spend afternoons on wild goose chases :))

**Comment by nklayman** [comment]
BTW, you convinced me. As soon as your PR gets merged I will use `electron-devtools-installer` instead of my custom version.

I thought your PR would be merged quickly, but since it hasn't I thought I should let you know what's up.

Thanks for your input and suggestion!
